src/standard_library/array_methods.md

```diff
@@ -0,0 +1,139 @@
+
+# Array
+
+For convenience, the STD provides some ready-to-use, common methods for arrays[^migrationNote]:
+
+## len
+
+Returns the length of an array
+
+```rust
+fn len(_array: Self) -> comptime Field
+```
+
+example
+
+```rust
+// arr = [42, 42]
+fn main(arr : pub [Field; 2]) {
+    let t = arr.len();
+    constrain t == 2;
```

Comment:
Nitpick: We wouldn't need the `arr = [42, 42]` comment if we used a literal to begin with:

```rs
fn main() {
    let array = [42, 42];
    constrain array.len() == 2;
}
```